lib/surface/live_view.ex

```diff
@@ -41,7 +41,7 @@ defmodule Surface.LiveView do
       @before_compile unquote(__MODULE__)
 
       @doc "The id of the live view"
-      property id, :integer
+      property id, :string, required: true
```

Comment:
I think the `id` of a Liveview is required only if it's a child Liveview. According to the docs:

> When rendering a child LiveView, the :id option is required to uniquely identify the child. A child LiveView will only ever be rendered and mounted a single time, provided its ID remains unchanged. Updates to a child session will be merged on the client, but not passed back up until either a crash and re-mount or a connection drop and recovery. To force a child to re-mount with new session data, a new ID must be provided.

If that's true, I think we should make it required only if we can check if there's a parent, i.e. it's not a root LV. Otherwise, we should just leave it optional.

Reply:
Won't it always be a child live view when our validation is run? I don't think we do any validation on the assigns when a `Sirfave.LiveView` is used from the router or a live_render call

Is your concern around the documentation we generate? Or validation that's too strict?

Reply:
> Won't it always be a child live view when our validation is run? I don't think we do any validation on the assigns when a `Sirfave.LiveView` is used from the router or a live_render call

You're probably right. If that's true, then my concern is unfounded. Feel free to merge it :)